app/src/services/sync.js

```diff
@@ -91,95 +138,66 @@ const service = {
     }
   },
 
-
   /**
    * @function syncObject
-   * syncs object-level data
-   * @param {string} [options.path] The path of object in sync job
-   * @param {string} [options.bucketId] The uuid bucketId of object in sync job
+   * Synchronizes Object level data
+   * @param {string} options.path The path of object in sync job
+   * @param {string | null} options.bucketId The uuid bucketId of object in sync job
    * @param {string} [options.userId=SYSTEM_USER] The uuid of a user that created the sync job
    * @param {object} [etrx=undefined] An optional Objection Transaction object
-   * @returns {object} synced objects that exist in COMS and S3 eg:
-   * <ObjectModel> (synced object)
-   * or `undefined` (when object was pruned from COMS db after sync)
+   * @returns {Promise<Array<object | undefined>>} Either an array of synced objects or undefined
```

Comment:
i think this syncObject method returns just one js object 
`@returns {Promise<object | undefined>}`